### PR TITLE
[IMP] web,calendar: improve design of popover for crowded events

### PR DESCRIPTION
**Pull request by vili-odoo**
The calendar popover for a crowded event is now easier to read and navigate.

task-2685440

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/calendar/static/src/views/fields/many2many_attendee.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <t t-name="calendar.Many2ManyAttendee" t-inherit="web.Many2ManyTagsAvatarField" t-inherit-mode="primary" owl="1">
+    </t>
+
+</templates>
```
Comment (leonardopavanrocha) [inline addons/calendar/static/src/views/fields/many2many_attendee.xml:7]: not needed maybe?

File: addons/calendar/static/src/views/fields/many2many_attendee_expandable.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        </xpath>
+        <xpath expr="//TagsList" position="before">
+            <t t-if="tags.length > 5">
+                <div class="d-flex w-100">
+                    <div class="flex-grow-1">
+                        <strong><t t-out="tags.length"/> attendees</strong>
+                        <t t-set="partners" t-value="props.record.preloadedData.partner_ids"/>
+                        <br/><t t-out="partners.filter(p => p.status === 'accepted').length"/> accepted
+                        <br/><t t-out="partners.filter(p => p.status === 'declined').length"/> declined
+                        <br/><t t-out="partners.filter(p => p.status !== 'accepted' &amp;&amp; p.status !== 'declined').length"/> uncertain
```
Comment (leonardopavanrocha) [inline addons/calendar/static/src/views/fields/many2many_attendee_expandable.xml:16]: Could be replaced by getters in the component level. For example

```js
export class Many2ManyAttendeeExpandable extends Many2ManyAttendee {
   get partners() {
        return this.props.record.preloadedData.partner_ids;
    }

    get acceptedPartnersCount() {
        return this.partners.filter(p => p.status === 'accepted').length;
    }
    ...
}
```
and in the template you just use 

```xml
    <t t-out="acceptedPartnersCount"/> accepted
```

File: addons/calendar/static/src/views/fields/many2many_attendee_expandable.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        useEffect(() => {
+            if (this.shouldUpdatePopoverPosition) {
+                this.shouldUpdatePopoverPosition = false;
+
+                const popover = document.querySelector(".o_field_many2manyattendeeexpandable")
+                    .closest(".o_popover");
+                const targetElement = document.querySelector(`.fc-event[data-event-id="${this.props.record.resId}"]`);
+                reposition(targetElement, popover, { position: "right", margin: 0 });
+            }
+        });
```
Comment (leonardopavanrocha) [inline addons/calendar/static/src/views/fields/many2many_attendee_expandable.js:29]: `useEffect` is a hook that is run on every re-render. It receives two parameters, a callback and a function that returns a list of dependencies. The dependencies are evaluated every re-render and if they change, the callback will be executed. So you can get rid of your `shouldUpdatePopoverPosition` property and you can just add `this.state.expanded` in your dependency array.

```suggestion
        useEffect(() => {
                const popover = document.querySelector(".o_field_many2manyattendeeexpandable")
                    .closest(".o_popover");
                const targetElement = document.querySelector(`.fc-event[data-event-id="${this.props.record.resId}"]`);
                reposition(targetElement, popover, { position: "right", margin: 0 });
            }
        }, () => [this.state.expanded]);
```

**Review by leonardopavanrocha — CHANGES_REQUESTED** [review]
Thanks for your work! I just added a small request that would improve the code.

**Review by leonardopavanrocha — APPROVED** [review]
LGTM

File: addons/calendar/static/src/views/fields/many2many_attendee_expandable.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <t t-name="calendar.Many2ManyAttendeeExpandable" t-inherit="web.Many2ManyTagsAvatarField" t-inherit-mode="primary" owl="1">
+        <xpath expr="div" position="attributes">
```
Comment (arj-odoo) [inline addons/calendar/static/src/views/fields/many2many_attendee_expandable.xml:5]: I would add an `id` to the div in web.Many2ManyTagsAvatarField and target it here (no need to use a xpath). 
```xml
<div id="my_id" position="attributes">
```

That way, you are more selective and there will be no ambiguity if the web.Many2ManyTagsAvatarField component evolves in the future.

**Review by arj-odoo — CHANGES_REQUESTED** [review]
Good work ! It will really improve the way we display these information. I have only one small remark for code maintainability.

**Comment by arj-odoo** [comment]
@robodoo r+

**Pull request closed without merging**